hazelcast/src/main/java/com/hazelcast/sql/impl/type/converter/AbstractTemporalConverter.java

```diff
@@ -46,6 +47,16 @@ protected static LocalDateTime dateToTimestamp(LocalDate date) {
         return date.atStartOfDay();
     }
 
+    /**
+     * Convert DATE to TIMESTAMP
+     *
+     * @param date Date.
+     * @return Timestamp with time zone at UTC midnight.
+     */
+    protected static OffsetDateTime dateToTimestampWithTimezone(LocalDate date) {
+        return OffsetDateTime.of(date, LocalTime.of(0, 0, 0), ZoneOffset.UTC);
```

Comment:
We must use the system default zone:
```suggestion
        return timestampToTimestampWithTimezone(dateToTimestamp(date));
```

Reply:
The suggestion is actually how it was implemented originally. I changed this to fix https://github.com/hazelcast/hazelcast/pull/18570/files#r620334467

It seems that my approach was wrong. It fixes one problem but causing another one. I think I need to think how `EXTRACT` works on different types of data instead of casting everything to `TIMESTAMPTZ`.

Reply:
Created a followup issue for this problem: #18620